stats/stats-server/src/charts.rs

```diff
@@ -38,8 +29,30 @@ fn new_hashset_check_duplicates<T: Hash + Eq, I: IntoIterator<Item = T>>(
     Ok(result)
 }
 
+struct ValidatedConfig {
+    charts: Vec<ArcChart>,
+    counters_filter: HashSet<String>,
+    lines_filter: HashSet<String>,
+}
+
 impl Charts {
     pub fn new(config: Config) -> Result<Self, anyhow::Error> {
+        let ValidatedConfig {
+            charts,
+            counters_filter,
+            lines_filter,
+        } = Self::validate_config(&config)?;
+        let settings = Self::new_settings(&config);
```

Comment:
maybe move `new_settings` logic into validate_config function, so that you don't need to split ValidatedConfig and Settings and also you don't need to assume that config is valid

Reply:
When I was developing this, I made everything in one function `new`. Then I decided to split it into two functions, validation and then some additional computations, like new settings. There is nothing wrong with assumption that config should be valid imo 
I don't think that putting everything in one big function is better, but idk